### PR TITLE
Pie charts polar

### PR DESCRIPTION
**Pull request by OliBravo**

### CODE REVIEW

File: R/tooltips.R [hunk 1]
```diff
@@ -190,3 +192,217 @@ body(saveAndGetTooltips)[[length(body(saveAndGetTooltips))]] <- quote(
...
+#' @inheritParams getTooltips
+#'
+#' @details Currently only single layer plots are handled
+#'
+#' @return
+#'
+getTooltipsForPiechart <- function(pp, varDict) {
+  if (length(pp$layers) > 1) {
+    warning("Only single layer plots are accepted")
+    return(NULL)
```
Comment (cosi1) [inline R/tooltips.R:209]: I wonder if we should "fail silently" here. What will happen here when someone runs this in their Shiny app? Will the tooltips be disabled (desired behavior)?

File: R/tooltips.R [hunk 2]
```diff
@@ -190,3 +192,217 @@ body(saveAndGetTooltips)[[length(body(saveAndGetTooltips))]] <- quote(
...
+  }
+
+  built = ggplot2::ggplot_build(pp)
```
Comment (cosi1) [inline R/tooltips.R:212]: Assignment operator ;-)

File: R/tooltips.R [hunk 3]
```diff
@@ -190,3 +192,217 @@ body(saveAndGetTooltips)[[length(body(saveAndGetTooltips))]] <- quote(
...
+                      x = grob_names,
+                      value = TRUE)
+  panel_positions <- grep(pattern = "^panel-[0-9]+\\..*",
+                          x = grob_names,
+                          value = FALSE)
+  list(names = panel_names, positions = panel_positions)
+}
+
+getPanelsBorders <- function(p) {
+  plot(p) # TODO: how to avoid plotting? removing this line causes a viewport error
```
Comment (cosi1) [inline R/tooltips.R:246]: I remember having a similar problem – I believe some function from `gridExtra` helped...

File: R/tooltips.R [hunk 4]
```diff
@@ -190,3 +192,217 @@ body(saveAndGetTooltips)[[length(body(saveAndGetTooltips))]] <- quote(
...
+                                         ifelse(x == 0 & y < 0,
+                                                3 * pi / 2,
+                                                0)))))
+
+      polars <- data.frame(x = x, y = y, r = r, phi = phi)
+      polars <- polars[!(polars$x == 0 & polars$y == 0), ]
+
+      phi_limits <- polars
+      phi_limits$phi <- round(phi_limits$phi, 6)
+      phi_limits <- aggregate(x ~ phi, data = phi_limits, length)
```
Comment (cosi1) [inline R/tooltips.R:335]: Is it from `{dplyr}`? Do we have it as a dependency? (Plus missing namespace.)

EDIT: Nevermind, I'm stupid.

**Review by cosi1 — APPROVED** [review]

File: R/tooltips.R (hunk 3 above)
Comment (OliBravo) [reply R/tooltips.R:246]: Will search for it

File: R/tooltips.R (hunk 2 above)
Comment (OliBravo) [reply R/tooltips.R:212]: uuuu...

File: R/tooltips.R (hunk 1 above)
Comment (OliBravo) [reply R/tooltips.R:209]: Maybe we should fail, maybe not. The app won't crash. It's not fully connected with front-end code so it's hard for me to fully comment the behavior presented here. It's rather a matter of agreement what should happen. Eventually, multilayer plots should be handled as well. The problem of tooltips for multilayer charts is not resolved at all currently and should be discussed separately.

File: R/tooltips.R (hunk 2 above)
Comment (OliBravo) [reply R/tooltips.R:212]: corrected

**Pull request merged**